homeassistant/helpers/entity_registry.py

```diff
@@ -1085,6 +1087,8 @@ def _async_update_entity(
         if not new_values:
             return old
 
+        self.hass.verify_event_loop_thread("_async_update_entity")
```

Comment:
This one is a bit later because its only actually unsafe if we are going to modify the data so we don't have to pay the cost of the check when its useless. `async_get_entity_id` is actually thread-safe.